src/app/GenericReadHandlerCallback.h

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include "ReadHandler.h"
+
+namespace chip {
+namespace app {
+
+class GenericReadHandlerCallback : public ReadHandler::ApplicationCallback
```

Comment:
This name is not representative of the behavior. The behavior should be explained in a class comment. The behavior of this class is to extend the max sleep interval to the SED idle interval if requested lower by the subscriber.

Suggest `SubscriptionMaxIntervalStretcher` as a class name.

Reply:
done